src/utils/cuda/cuda_utils.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* forward declaration for the internal CUDA data structure */
+
+#include <nixl.h>
+#include <memory>
+
+/****************************************
+ * Pointer Context
+*****************************************/
+
+namespace nixlCuda {
+    class memCtx {
+    public:
+        enum memory_t {
+            MEM_NONE,
+            MEM_HOST,
+            MEM_DEV,
+            MEM_VMM_DEV,
+        } ;
+    protected:
+        memory_t memType;
+    public:
+        memCtx() : memType(MEM_NONE)
+        {  }
+
+        virtual ~memCtx() = default;
+
+        memory_t getMemType() const {
```

Comment:
This method seems to be redundant for this class, and can be removed. `memType` can be moved to the implementation along with these changes to improve encapsulation of the class fields.

Reply:
This is used outside, specifically in the unit test now.
As it's not a complex functionality, I would like to keep it

Reply:
I will address it as @ColinNV suggested by returning a constant value instead of storing it in the variable

Reply:
Thanks!

However, I do not agree that the use in the test is the reason for adding it to class interface. We should keep any class as as concise as possible. And `enum memory_t` can also be moved to implementation.
The combination of `enableAddr` (to detect context) and `set` (to set context) seems to be sufficient for a user of the class. The detected memory type is an internal detail of the implementation.

Reply:
This is effectively the only way of verifying that the class is doing the right thing:
https://github.com/ai-dynamo/nixl/pull/227/files#diff-14c355505ca3f0b041657941c7a4ab04fcc252f1cc6c7ed55b15847b3e493582R218

IMO it is a very minor sacrifice for the opportunity to have a way to test it.
As an option, an additional friend class can be defined serving as a side-channel for the test.
IMO it's too much of a workaround.

Reply:
I don't think that exposing implementation details is the right way to test the interface of the class. I think it would be better to do the following.
1. bind cuda context to the cpu thread (using Runtime API with cudaSetDevice or Driver API with cuCtxSetCurrent)
2. allocate memory (cudaMalloc, VMM API)
3. pop context (or use std::thread)
4. call enableAddr
5. call set
6. check that current context equal to the one that set before the allocation

In this case, the test does not depend on the implementation.

Reply:
ok that works

Reply:
this is still TODO

Reply:
Done